src/nodes.rs

```diff
@@ -4,13 +4,19 @@ use std::collections::{BTreeSet, HashMap, HashSet};
 #[derive(Default, Debug)]
 pub struct Nodes {
     all: HashMap<Id, Node>,
-
     quarantined: HashSet<Id>,
-
     not_reachable: BTreeSet<Id>,
     available: BTreeSet<Id>,
 }
 
+#[derive(Clone, Debug)]
```

Comment:
It would make sense to also add the support for `Serde` so you won't have to do a conversion type on the other side (for the REST API).

```suggestion
#[derive(Clone, Debug, Serialize)]
```